corehq/ex-submodules/casexml/apps/case/models.py

```diff
@@ -604,7 +606,19 @@ def apply_updates(self, update_action):
         Applies updates to a case
         """
         for k, v in update_action.updated_known_properties.items():
-            setattr(self, k, v)
+            try:
+                setattr(self, k, v)
+            except BadValueError as e:
+                soft_assert(notify_admins=True)(False, u'Error applying {}: {}'.format(
+                    unicode(update_action), e
+                ))
+                # todo: awkward hard-coded handling for legacy forms - can remove if these
+                # assertions go away: http://manage.dimagi.com/default.asp?182759
+                if k == 'opened_on' and isinstance(v, basestring):
+                    # this will reraise the exception if this wasn't a string date
+                    setattr(self, k, iso_string_to_datetime(v))
```

Comment:
Nice :p

On Wed, Sep 30, 2015 at 12:04 PM Cory Zue <notifications@github.com> wrote:

> In corehq/ex-submodules/casexml/apps/case/models.py
> <https://github.com/dimagi/commcare-hq/pull/8416#discussion_r40813641>:
>
> > @@ -604,7 +606,19 @@ def apply_updates(self, update_action):
> >          Applies updates to a case
> >          """
> >          for k, v in update_action.updated_known_properties.items():
> > -            setattr(self, k, v)
> > +            try:
> > +                setattr(self, k, v)
> > +            except BadValueError as e:
> > +                soft_assert(notify_admins=True)(False, u'Error applying {}: {}'.format(
> > +                    unicode(update_action), e
> > +                ))
> > +                # todo: awkward hard-coded handling for legacy forms - can remove if these
> > +                # assertions go away: http://manage.dimagi.com/default.asp?182759
> > +                if k == 'opened_on' and isinstance(v, basestring):
> > +                    # this will reraise the exception if this wasn't a string date
> > +                    setattr(self, k, iso_string_to_datetime(v))
>
> yep exactly
> <https://github.com/dimagi/commcare-hq/blob/master/corehq/ex-submodules/casexml/apps/case/xml/parser.py#L113>.
> who wrote this crap anyways?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/8416/files#r40813641>.
>